### ISSUE TITLE
Modern Search: Search box with a "Clear" option

### ISSUE THREAD

**Issue by coupalm** [SharePoint/sp-dev-solutions#119]
#### Category
- [ ] Question
- [ ] Bug
- [x ] Enhancement

#### Expected or Desired Behavior
When the search results web part is using a default search query (ex: *), the web part will automatically display results. Then, if the user enters a search criteria in the search box web part, the results will be returned. If he wants to start again and see the same default results, he could click on a "Clear" search box option that would return the default search query results.

#### Observed Behavior
If the user clears the search box info and hit enter, nothing will happend in the search results because he would needs to enter * (in my example) in the search box to get the same default search query. 

It would be more intuitive to get a "Clear" option.

Thanks for the BEST added value modern web part so far!

**Comment by YannickRe** [comment]
We mentioned this in #72 too, and I need it too so as soon as I have some time I will put it in 😉

**Comment by coupalm** [comment]
Working great! Thanks.

**Comment by mag210** [comment]
I wish I'd seen this before implementing the same improvement lol! Looks good. One possible suggestion would be to also reset to the default search if the user simply deletes his query.

**Comment by YannickRe** [comment]
@mag210 We were doing a lot of improvements in a short time, with multiple people, so currently we target a dev branch. This might be the reason you missed it.
Not sure when @wobba plans to release to master.

Just clearing feels “weird” to me: clear + enter, clicking clear button or hitting escape key feel more natural. We also need to hit enter to execute a normal query.

**Comment by wobba** [comment]
We could do monthly releases, or I'll merge when you @YannickRe and @FranckyC feels it's stable enough :)

**Comment by mag210** [comment]
@YannickRe  thanks for this. I think the roadmap feature is a good idea, and maybe needs to show on the Readme so developers can see what is coming next and whether anything they might want to do is already in the pipeline for the community. Monthly release would seem to be the way to go?

In terms of the clearing behavior, I guess depends on personal preference / what users are use to. I used the existing modern site search bar as a guide when implementing as I figured this would be what most users of modern SharePoint would interact with. It does clear results when you delete your query and returns to Microsoft's default "0" type search suggestions.  Equally the search icon only appears when the user has typed something in indicating they can now search. Ultimately, I  guess it probably depends on how the webpart will be implemented as to which approach makes more sense. Anyway, I should say that I think this webpart(s) is the best SPFX solution I've seen! Great work, and happy to help with any testing / future iterations. It fills so many gaps in the currently very limited modern search experience.
